src/TestProblem/ELBDM/DiskHeating/Init_TestProb_ELBDM_DiskHeating.cpp

```diff
@@ -0,0 +1,1208 @@
+#     endif
+
+      // read velocity dispersion table
+      if ( DispNbin == 0){
+         if ( MPI_Rank == 0 )    Aux_Error( ERROR_INFO, "DispNbin should be set!\n" );
+      }
+      int STR_SIZE = 1024;
+      int counter, k;
+      FILE *input_table;
+      char buff[STR_SIZE];
+      double radius, value;
+      DispTable_r = new double [DispNbin];
+      DispTable_d = new double [DispNbin];
+
+      // velocity dispersion table for thin disk
+      if ( MPI_Rank == 0 )
+      {
+         counter = 0;
+         input_table = fopen(DispTableFile,"r");
+         if (!input_table)
+         Aux_Error( ERROR_INFO, "File %s cannot be opened! Exit!\n", DispTableFile);
+         else
+            Aux_Message( stdout, "Loading %s for reading density table succeed!\n", DispTableFile);
+         while (!feof(input_table))
+         {
+            fgets(buff, STR_SIZE, input_table);
+            if (buff==NULL||buff[0]=='\0')
+               break;
+            else if (buff[0]!='#')
+            {
+               sscanf(buff, "%lf %lf", &radius, &value);
+               DispTable_r[counter] = radius;
+               DispTable_d[counter] = value*1e5/UNIT_V;
+               counter ++;
+            }
+            memset(buff, '\0', STR_SIZE);
+         }
+         fclose(input_table);
+      }
+      MPI_Bcast(DispTable_r, DispNbin, MPI_DOUBLE, 0, MPI_COMM_WORLD);
+      MPI_Bcast(DispTable_d, DispNbin, MPI_DOUBLE, 0, MPI_COMM_WORLD);
+
+      if ( MPI_Rank == 0 )
+      {
+         Aux_Message( stdout, "%s ...\n", __FUNCTION__ );
+
+         const double ParM = Disk_Mass / NPar_AllRank;
+         double Ran, RanR, RanV, R, Rold, f, f_, phi, RanVec[3];
+         Aux_Message(stdout, " Particle Mass = %13.7e\n", ParM) ;
+
+//       initialize the RNG
+         RNG = new RandomNumber_t( 1 );
+         RNG->SetSeed( 0, 1002 );
+
+         for ( long p = 0; p < NPar_AllRank; p++)
+         {
+            Time_AllRank[p] = Time[0];
+//          mass
+            Mass_AllRank[p] = ParM;
+//          label
+            Type_AllRank[p] = 3;
+            Label_AllRank[p] = 3;
+
+//          position
+            Ran  = RNG->GetValue( 0, 0.0, 1.0);
+            R = 1.0;
+            do
+            {
+               f = (1 + R) * exp(-R) + Ran - 1;
+               f_ = -R * exp(-R);
+               Rold = R;
+               R = R - f / f_;
+            }
+            while(fabs(R - Rold) / R > 1e-7);
+            RanR = Disk_R*R;
+            phi = 2*pi*RNG->GetValue( 0, 0.0, 1.0 );
+            RanVec[0] = RanR*cos(phi);
+            RanVec[1] = RanR*sin(phi);
+            RanVec[2] = 0;
+            for (int d = 0; d < 3; d++) Pos_AllRank[d][p] = RanVec[d] + Cen[d];
+
+//          velocity
+            double V_acc, sigma, r;
+            double RanV[3];
+            V_acc = sqrt(Ran*Disk_Mass/RanR);
+            Vel_AllRank[0][p] = - V_acc*sin(phi);
+            Vel_AllRank[1][p] =   V_acc*cos(phi);
+            Vel_AllRank[2][p] =   0;
```

Comment:
IIUC, it will be overwritten by `Par_AfterAcceleration_Ptr()`. Should we remove these lines?